pandas/tslib.pyx

```diff
@@ -3849,6 +3849,7 @@ def get_time_micros(ndarray[int64_t] dtindex):
 
 
 @cython.wraparound(False)
+@cython.boundscheck(False)
 def get_date_field(ndarray[int64_t] dtindex, object field):
```

Comment:
If you declared `field` as `char[:]` instead would you be able to `nogil` the whole thing until `raise`?